### DISCUSSION TITLE
What is "ortho (vert/horiz) downscaling (bicubic)" and how can I change it from bicubic? or does it matter?

### DISCUSSION THREAD

**Discussion by marat569** [mpv-player/mpv#14298] [Q&A] [answered]
So I was watching a 3840x1600 movie on my 3840x2160 monitor, and noticed I had two shaders active:

ortho (vert/horiz) downscaling (bicubic) 

I have no idea where they're from, scale/dscale/cscale are `ewa_lanczos4sharpest`

mpy mpv.conf: <https://github.com/marat569/mpv.conf/blob/main/mpv.conf>

so two questions:
1) what is that ortho thing doing
2) does it matter that its bicubic? I have performance to spare for something fancier than bicubic 

When I googled the effect, I found only github issue with no solid solution

Also shoutout to Kasper93 -- my dude helps out with every post and issue

![image](https://github.com/mpv-player/mpv/assets/7017110/d4f9203e-8c72-4c1a-a52a-633c4a918206)
![image](https://github.com/mpv-player/mpv/assets/7017110/35aac91f-aa4b-4120-a7a2-ce4b969e49a1)

**Comment by hooke007** [comment, marked as answer]
dup https://github.com/mpv-player/mpv/issues/11841

**Reply by marat569** [reply]
> dup #11841

So quick question:

1) Is bicubic for the two scalers perfectly fine? or is there a conf flag I can use to use something fancier?

Like I don't mind the scaling being there, my question is the fact that its using bicubic instead of something nicer -- unless whatever its doing doesn't really need anything fancy

**Reply by hooke007** [reply]
You'd better ask in that issue instead. DEVS always know more than users.

**Reply by marat569** [reply]
copy will do

**Comment by ghost** [comment]
b-spline (bicubic) is a smoothing filter and its only applied to the luma plane for feature extraction. it has no effect on chroma/rgb scaling and it's hard-coded, so you can't change it even if you wanted to. the only thing it does is map out the high-frequency and low-frequency information on the luma plane.

most feature extraction algorithms typically use either nearest neighbor, bilinear or b-spline. and out of those three b-spline is smoother and higher quality for feature extraction because it preserves source details while also not producing any unnatural jagged edges.

**Reply by marat569** [reply]
copy

I assume if its hardcoded, the impact it has on the image is negligible vs. something more "heavy"

**Reply by ghost** [reply]
bicubic is already computationally more "heavy" than both nearest-neighbor and bilinear.

**Reply by marat569** [reply]
> bicubic is already computationally more "heavy" than both nearest-neighbor and bilinear.

What about the quality

I'm trying to say heavier as in looks better ( since stuff that looks better tends to be heavier ;p )

**Reply by ghost** [reply]
like I already said, bicubic produces a higher quality feature map than the other aforementioned interpolation methods. its just blending the low-frequency information to extract out the high-frequency details.

**Reply by marat569** [reply]
copy, I misread mb !

well I have no issue with the functionality then! if there is no quality loss, I trust mpv to do its thang!
